src/oss-find-squats-lib/MutateExtension.cs

```diff
@@ -105,22 +105,25 @@ public static class MutateExtension
 
             // Check to see if it is a scoped npm package to generate candidates for.
             bool isScoped = purl.Namespace.IsNotBlank() && purl.Type.Equals("npm", StringComparison.OrdinalIgnoreCase);
-            string nameToMutate = isScoped ? (purl.Namespace!).Substring(1) : purl.Name;
+            string nameToMutate = isScoped ? purl.Namespace : purl.Name;
 
             foreach (IMutator mutator in mutators)
             {
                 foreach (Mutation mutation in mutator.Generate(nameToMutate))
                 {
                     // Construct the mutated name if the package was scoped.
-                    string mutated = isScoped ? $"@{mutation.Mutated}/{purl.Name}" : mutation.Mutated;
+                    string mutated = isScoped ? $"{mutation.Mutated}/{purl.Name}" : mutation.Mutated;
+                    string original = isScoped ? $"{purl.Namespace}/{purl.Name}" : purl.Name;
 
+                    Mutation calculatedMutation = new(mutated, original, mutation.Reason, mutation.Mutator);
+                    
                     if (generatedMutations.ContainsKey(mutated))
```

Comment:
nit: this code can be simplified by having an if statement that checks if the dictionary entry exists, adding it if it does not, and then adding to that list outside of the if statement.